src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTranslationAction.Actions.cs

```diff
@@ -124,11 +127,76 @@ public ProjectAssemblyNameAction(string assemblyName)
                     _assemblyName = assemblyName;
                 }
 
-                public override Task<Compilation> InvokeAsync(Compilation oldCompilation, CancellationToken cancellationToken)
+                public override Task<Compilation> TransformCompilationAsync(Compilation oldCompilation, CancellationToken cancellationToken)
                 {
                     return Task.FromResult(oldCompilation.WithAssemblyName(_assemblyName));
                 }
             }
+
+            internal sealed class AddAnalyzerReferencesAction : CompilationTranslationAction
+            {
+                private readonly IEnumerable<AnalyzerReference> _analyzerReferences;
+
+                public AddAnalyzerReferencesAction(IEnumerable<AnalyzerReference> analyzerReferences)
```

Comment:
>IEnumerable<AnalyzerReference> analyzerReferences [](start = 51, length = 49)

I'd say these should be ImmutableArrays, but for that you'd need my PR to be merged first :)

Reply:
yes, i would strongly want these not be IEnumerables.  can that not be done in this PR?

Reply:
@tmat merged in #42451 so I'll get that over to the base branch so I can use the new helpers added.